lib/scripts/iob_module.py

```diff
@@ -16,40 +17,40 @@
 from iob_verilog_instance import iob_verilog_instance
 
 
+@dataclass
 class iob_module:
-    """Generic class to describe a base iob-module"""
-
-    def __init__(self, name=None):
-        self.name = name or self.__class__.__name__
-        self.csr_if = "iob"
-        self.version = "1.0"  # Module version
-        self.description = "default description"  # Module description
-        self.previous_version = self.version  # Module previous version
-        self.setup_dir = ""  # Setup directory for this module
-        self.build_dir = ""  # Build directory for this module
-        self.rw_overlap = False  # overlap Read and Write register addresses
-        self.is_top_module = False  # Select if this module is the top module
-        self.use_netlist = False  # use module netlist
-        self.is_system = False  # create software files in build directory
-        self.board_list = None  # List of fpga files to copy to build directory
-        self.purpose = "hardware"
-        self.confs = []
-        self.regs = []
-        self.ios = []
-        self.block_groups = []
-        self.submodule_list = []
-        self.ignore_snippets = []  # List of snippets to ignore during replace
-
-        # Read-only dictionary with relation between the setup_purpose and the corresponding source folder
-        self.PURPOSE_DIRS = {
-            "hardware": "hardware/src",
-            "simulation": "hardware/simulation/src",
-            "fpga": "hardware/fpga/src",
-        }
-
-    def _setup(self, is_top=True, purpose="hardware", topdir="."):
+    """Generic class to describe how to generate a base IOb IP core / module"""
+
+    name: str = name or self.__class__.__name__
```

Comment:
This line throws a syntax error when using dataclasses for two reasons:
1) The keyword `self` is not recognized. Probably can be replaced by just `__class__.__name__`
2) The `name` variable is not recognized because it has not been defined before. This variable was previously an argument o the `__init__` method. Now it can probably be removed.

Reply:
This probably makes it work as intended (by default use the class name unless a subclass specifies a new name):
```suggestion
    name: str = __class__.__name__
```

Reply:
This did not work, the `__class__` variable was not recognized. I had to move this to the `__post_init__()` function [here](https://github.com/IObundle/iob-soc/pull/798/commits/d23aafbef7daf857c44145c6bb83cc3debb99426#diff-7db188bf8175e04672ebfa1ed4a534bcfde732969599f553027f872b0c845911R57).